File: src/main/java/com/khg/info_master/domain/Answer.java
package com.khg.info_master.domain;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "answer")
public class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "question_id", nullable = false, unique = true)
    private Question question;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String answerText;

    private Integer score;
    private String comment;
}

File: info-master-front/src/types/Answer.ts
// 서버에서 내려오는 답안 DTO
export interface Answer {
  id?: number;          // 새로 생성 시 undefined
  questionId: number;   // 어떤 문제의 답안인지
  answerText: string;

  // 선택 메타 (나중에 쓰고 싶으면)
  score?: number | null;
  comment?: string | null;

  createdAt?: string;
  updatedAt?: string;
}
